modules/howtos/examples/transactions-example.php

```diff
@@ -0,0 +1,489 @@
+// tag::create[]
+try {
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) {
+    // `$ctx` is a TransactionAttemptContext which permits getting, inserting,
+    // removing and replacing documents, performing N1QL queries, etc.
+    // … Your transaction logic here …
+    // Committing is implicit at the end of the lambda.
+    });
+}
+catch (\Couchbase\Exception\TransactionOperationFailedException $e) {
+    echo "Transaction did not reach commit point: $e\n";
+}
+catch (\Couchbase\Exception\TransactionException $e) { 
+// TODO check is this equivalent to TransactionCommitAmbiguousError as per Node examples?
+  echo "Transaction possibly committed: $e\n";
+}
+// end::create[]
+
+// tag::examples[]
+
+try {
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection) {
+      // Inserting a doc:
+      $ctx->insert($collection, 'doc-a', []);
+
+      // Getting documents:
+      $docA = $ctx->get($collection, 'doc-a');
+
+      // Replacing a doc:
+      $docB = $ctx->get($collection, 'doc-b');
+      $content = $docB->content;
+      $newContent = array_merge(
+        ["transactions" => "are awesome"],
+        $content);
+      
+      $ctx->replace($docB, $newContent);
+
+      // Removing a doc:
+      $docC = $ctx->get($collection, 'doc-c');
+      $ctx->remove($docC);
+
+      // Performing a SELECT N1QL query against a scope:
+      $qr = $ctx->query('SELECT * FROM hotel WHERE country = $1',
+        [ "scope" => "inventory",
+          "parameters" => ["United Kingdom"]
+      ]);
+      echo $qr->rows();
+
+      $ctx->query('UPDATE route SET airlineid = $1 WHERE airline = $2',
+        [ "scope" => "inventory",
+          "parameters" => ['airline_137', 'AF'] ] );
+    });
+}
+catch (\Couchbase\Exception\TransactionOperationFailedException $e) {
+    echo "Transaction did not reach commit point: $e\n";
+}
+catch (\Couchbase\Exception\TransactionException $e) { 
+// TODO check is this equivalent to TransactionCommitAmbiguousError as per Node examples?
+  echo "Transaction possibly committed: $e\n";
+}
+// end::examples[]
+
+// execute other examples
+getExample($cluster, $collection);
+getReadOwnWritesExample($cluster, $collection);
+replaceExample($cluster, $collection);
+removeExample($cluster, $collection);
+insertExample($cluster, $collection);
+queryExamples($cluster, $collection);
+queryInsert($cluster, $collection);
+queryRYOW($cluster);
+queryOptions($cluster);
+querySingle($cluster);
+
+# TODO set data up to run these examples fully
+# playerHitsMonster(42, "arthur", "vogon", $cluster, $collection);
+# rollback(...)
+# rollbackCause(...)
+
+function getExample($cluster, $collection) {
+  echo "getExample\n";
+  // tag::get[]
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection) { 
+      $docA = $ctx->get($collection, "doc-a");
+    });
+  // end::get[]
+  // TODO: should this show nullable/optional in an example?
```

Comment:
As of SDK-integration we now have all implementations raising DocumentNotFoundException from ctx->get(), same as SDK3.  So no nullable/optional ctx.getOptional() anymore.
And yes would be good to have an example of that.

Reply:
I'm confused: we want an example of Optional *not* existing here?
Ah, or of the DocumentNotFoundException?

Reply:
Yes, the latter.
And a double-check that getOptional is not in the API.